src/test/java/org/isf/utils/file/TestFileTools.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Open Hospital (www.open-hospital.org)
+ * Copyright © 2006-2021 Informatici Senza Frontiere (info@informaticisenzafrontiere.org)
+ *
+ * Open Hospital is a free and open source software for healthcare data management.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * https://www.gnu.org/licenses/gpl-3.0-standalone.html
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.isf.utils.file;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.TimeZone;
+
+import org.junit.Test;
+
+public class TestFileTools {
+
+	@Test
+	public void testGetTimestampFromName() throws Exception {
```

Comment:
the aim of the method is to correctly parse a date from tipical patterns in filenames (like from cameras, screenshots, downloads, ecc...). To test if the date also is correct is good, thus obvious.

Reply:
I think I understand the purpose but not so the implementation.  For example, parsing "2021-12-22 1100" returns four dates: 
````
Wed Dec 22 00:00:00 EST 2021
Wed Dec 21 00:00:00 EST 2022
Wed Dec 22 11:00:00 EST 2021
Wed Dec 22 11:00:00 EST 2021
````
I am not quite sure how that second entry matches to original string.

Reply:
It look like a bug... the entries are driven by the `dateTimeFormats` matrix, where I can see now there is a duplicated pattern `yyyy-MM-dd HHmm`, although this does not explain why the second match appear...

The idea is to prompt to the user a set of parsed data in order to choose one...

Reply:
Removing the duplicated pattern reduces the output to:
````
Wed Dec 22 00:00:00 EST 2021
Wed Dec 21 00:00:00 EST 2022
Wed Dec 22 11:00:00 EST 2021
````
which is better but still confusing.

So the first entry is returned because it matches:
````
{ "yyyy-MM-dd", "\\d{4}-\\d{2}-\\d{2}" },
````
The third entry is returned because it matches:
````
{ "yyyy-MM-dd HHmm", "\\d{4}-\\d{2}-\\d{2} \\d{4}" },
````
The second entry is returned be it matches:
````
{ "dd-MM-yy", "\\d{2}-\\d{2}-\\d{2}" },
````

It appears that `match` is matching substrings.  If we add `\\z" (which means the end of the input string) to the end of all the regular expressions I think it will fix the problem.

Reply:
I also removed the duplicated pattern and added a missing one: `dd-MM-yyyy HHmm`.

Reply:
Actually (20)21-12-22 is also a correct date! Good catch!
Nevertheless, adding end of line is not correct because most of the time the timestamp is at the beginning of the filename, followed by some other sequence of characters, thus not always true... definitely the date could be at any position. 

I don't master regex but maybe we can exclude any other digit [1-9] before and after each regex?